examples/property_prediction/pretrain_gnns/chem/pretrain_supervised.py

```diff
@@ -0,0 +1,144 @@
+import argparse
+import pickle
+import tqdm
+import csv
+from functools import partial
+import itertools
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.utils.data import DataLoader
+
+import dgl
+from rdkit import Chem
+from dgllife.utils import PretrainAtomFeaturizer
+from dgllife.utils import PretrainBondFeaturizer
+from dgllife.utils import smiles_to_bigraph
+from dgllife.model.model_zoo.gin_predictor import GINPredictor
+
+from utils import PretrainSupervisedMoleculeCSVDataset
+
+
+def collate(samples):
+    graphs, labels = map(list, zip(*samples))
+    batched_graph = dgl.batch(graphs)
+    batched_labels = torch.stack(labels)
+    return batched_graph, batched_labels
+
+
+def train(args, model, train_dataloader, optimizer, criterion, device):
+    for epoch in range(1, args.epochs):
+        model.train()
+        with tqdm.tqdm(train_dataloader) as tq_train:
+            for step, (bg, labels) in enumerate(tq_train):
+                node_feats = [
+                    bg.ndata.pop('atomic_number'),
+                    bg.ndata.pop('chirality_type')
+                ]
+                edge_feats = [
+                    bg.edata.pop('bond_type'),
+                    bg.edata.pop('bond_direction_type')
+                ]
+                bg = bg.to(device)
+                node_feats = [n.to(device) for n in node_feats]
+                edge_feats = [e.to(device) for e in edge_feats]
+
+                logits = model(bg, node_feats, edge_feats)
+
+                is_valid = labels ** 2 > 0
```

Comment:
Add a note about the possible values in labels and their meanings. Otherwise `** 2 > 0` can be confusing as happened to you.

Reply:
ty